testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -259,6 +261,44 @@ public void testReferencedTablesAndRoutines()
         assertEquals(routine.getAuthorization(), "user");
     }
 
+    @Test
+    public void testReferencedColumns()
```

Comment:
Would it be possible to add test to `TestAnalyzer`. It seems odd that we use listener to test analysis.

Reply:
The test I added is related to the original issue: https://github.com/trinodb/trino/issues/6960
I don't think that `TestAnalyzer`'s methods expose the Analysis, which would be necessary to verify how it identifies columns.